docx/oxml/text/run.py

```diff
@@ -136,36 +136,40 @@ def text(self):
         child elements like ``<w:tab/>`` translated to their Python
         equivalent.
         """
-        text = ''
-        if CT_FldChar.numOfNestedFldChar > 0:
-            for child in self:
-                if child.tag == qn('w:fldChar'):
-                    if child.fldCharType == 'begin':
-                        CT_FldChar.numOfNestedFldChar += 1
-                    else:
-                        # `w:fldChar` stores instruction text from `w:fldCharType='begin'`
-                        # to `w:fldCharType='separate'` if 'separate' exists, if not then
-                        # until `w:fldCharType='end'`.
-                        CT_FldChar.numOfNestedFldChar -= 1
-        else:
-            for child in self:
-                if child.tag == qn('w:t'):
-                    t_text = child.text
-                    text += t_text if t_text is not None else ''
-                elif child.tag == qn('w:tab'):
-                    text += '\t'
-                elif child.tag == qn('w:br'):
-                    text += '\n'
-                elif child.tag == qn('w:cr'):
-                    text += '\r'
-                elif child.tag == qn('w:noBreakHyphen'):
-                    text += '-'
-                elif child.tag == qn('w:sym'):
-                    text += child.readSymbol
-                elif child.tag == qn('w:fldChar'):
-                    if child.fldCharType == 'begin':
-                        CT_FldChar.numOfNestedFldChar += 1
-        return text
+        text = {'val':''}
```

Comment:
It's a dictionary so that the ineer function `appendText` could concatenate string. When it was a string variable it couldn't be referenced inside the `appendText` function.

Reply:
Makes sense, thanks for clarifying. I would suggest trying to avoid additional function in general, let me know what do you think of the changes I proposed below.